builder/vmware/iso/step_create_vmx.go

```diff
@@ -155,7 +156,34 @@ func (s *stepCreateVMX) Run(state multistep.StateBag) multistep.StepAction {
 	return multistep.ActionContinue
 }
 
-func (s *stepCreateVMX) Cleanup(multistep.StateBag) {
+func (s *stepCreateVMX) Cleanup(state multistep.StateBag) {
+	driver := state.Get("driver").(vmwcommon.Driver)
+	config := state.Get("config").(*Config)
+	ui := state.Get("ui").(packer.Ui)
+
+	ui.Say("Unregistering and deleting virtual machine...")
+	var err error = nil
+	for i := 0; i < 5; i++ {
+		running, err := driver.IsRunning(config.VMXTemplatePath)
+		if err == nil {
+			if running {
+				err = driver.Stop(config.VMXTemplatePath)
+			}
+		}
+		if err == nil {
+			time.Sleep(5 * time.Second * time.Duration(i))
```

Comment:
Understood. Just so I follow along, it will first sleep 0, then 5, then 10, then 15, and finally 20 seconds to wait for the shutdown. (total 50s). I haven't run this command manually so I don't know what's reasonable to expect, but seems OK to me.